test/integration-tests/test-server/test-server.ts

```diff
@@ -17,18 +17,18 @@ export const connectionOptions: ConnectionOptions = {
 };
 
 const customContext = (context: ExpressContext): Partial<TestContext> => {
-    const { req } = context;
+    const { req, connection } = context;
+    const headers = req ? req.headers : connection?.context;
```

Comment:
Context of connection is just headers?

Reply:
Yes, take a look [here](https://www.apollographql.com/docs/apollo-server/data/subscriptions/#context-with-subscriptions)